lib/log.dart

```diff
@@ -30,3 +32,51 @@ bool debugLog(String message) {
   }());
   return true;
 }
+
+typedef ReportErrorCallback = void Function(String? message, {String? details});
+/// Display an error message in a [SnackBar].
+///
+/// This shows a [SnackBar] containing the message if [ZulipApp] is ready,
+/// otherwise logs it to the console.
+///
+/// If `message` is null, this will clear the existing [SnackBar]s if there
+/// are any.  Useful for promptly dismissing errors.
+///
+/// If `details` is non-null, the [SnackBar] will contain a button that would
+/// open a dialog containing the error details.
+// This gets set in [ZulipApp].  We need this indirection to keep `lib/log.dart`
+// from importing widget code, because the file is a dependency for the rest of
+// the app.
+ReportErrorCallback get reportErrorToUser => (message, {details}) {
+  assert(() {
+    _debugLastReportedError = details != null ? '$message\n$details' : message;
+    return true;
+  }());
+  (_reportErrorToUser ??= _defaultReportErrorToUser)(message, details: details);
+};
+ReportErrorCallback? _reportErrorToUser;
+set reportErrorToUser(ReportErrorCallback? value) =>
+  _reportErrorToUser = value;
```

Comment:
nit: use block body

Generally a setter always gets a block body.

See https://github.com/flutter/flutter/blob/main/docs/contributing/Style-guide-for-Flutter-repo.md#consider-using--for-short-functions-and-methods